python/instrumentation/openinference-instrumentation-agno/src/openinference/instrumentation/agno/__init__.py

```diff
@@ -0,0 +1,213 @@
+from typing import Any, Callable, Collection, List, Optional, Type
+
+from opentelemetry import trace as trace_api
+from opentelemetry.instrumentation.instrumentor import BaseInstrumentor  # type: ignore
+from wrapt import wrap_function_wrapper
+
+from openinference.instrumentation import (
+    OITracer,
+    TraceConfig,
+)
+from openinference.instrumentation.agno._wrappers import (
+    _FunctionCallWrapper,
+    _ModelWrapper,
+    _RunWrapper,
+)
+from openinference.instrumentation.agno.version import __version__
+
+_instruments = ("agno >= 1.4.5",)
+
+
+# Find all model classes in agno.models that inherit from BaseModel
+def find_model_subclasses() -> List[Type[Any]]:
```

Comment:
I'm not 100% sure but this could be causing a fair amount of duplicate span nesting.
<img width="592" alt="Screenshot 2025-05-11 at 11 54 07 AM" src="https://github.com/user-attachments/assets/b07ecd9d-3e2a-4371-a54c-fd0977ec81ab" />

Reply:
Ah that is not the idea, it shouldn't be nested. There can be multiple invoke calls though, like if you make a request (invoke #1), the model responds with a tool call, the tool is executed and the result is sent to the model (invoke #2)